basilaslam/Angular-ecommerce
# implemented address page with all cred operations

## Pull Request: Address Functionality Implementation

### Issue
Fixes #4 - Address Functionality Implementation

### Description
This pull request addresses the issue mentioned in #4 by implementing the address functionality within the application.

### Changes Made
- Created the `AddressComponent` to handle address-related functionality.
- Implemented the `AddressCardComponent` to display address information.
- Developed the `AddressFormComponent` for editing and updating address details.
- Added the `ListAllAddressComponent` to list all available addresses.
- Integrated the necessary services and models to support the address features.
- Created required routes and navigation links for seamless user experience.
- Implemented reactive forms for a better user input experience in the `AddressFormComponent`.

### How to Test
1. Navigate to the address list page by using "/address/list-all" link in the navigation menu.
2. View all addresses listed on the page.
3. Click on the "Edit" button next to an address to modify its details.
4. Update the address details using the provided form.
5. Click the "Save" button to confirm the changes.
6. Verify that the updated address information is reflected in the list.

### Additional Notes
- The implemented address functionality follows the established UI and design patterns of the application.
- This implementation lays the foundation for further enhancements and features related to address management.

### Checklist
- [x] Tested locally and verified functionality.

### Related Issues
Closes #4
